warehouse/oidc/models/github.py

```diff
@@ -39,21 +37,17 @@ def _check_job_workflow_ref(ground_truth, signed_claim, all_signed_claims):
     if not signed_claim:
         raise InvalidPublisherError("The job_workflow_ref claim is empty")
 
+    # We need at least one of these to be non-empty
     ref = all_signed_claims.get("ref")
-    if ref is None:
-        raise InvalidPublisherError("The ref claim is missing")
-
-    if ref == "":
-        with sentry_sdk.push_scope() as scope:
-            scope.fingerprint = all_signed_claims["sub"]
-            sentry_sdk.capture_message(
-                "GitHub JWT has empty-string ref claim, other claims are: "
-                f"{all_signed_claims}"
-            )
+    sha = all_signed_claims.get("sha")
+    if not (ref or sha):
+        raise InvalidPublisherError("The ref and sha claims are empty")
 
-    if not (expected := f"{ground_truth}@{ref}") == signed_claim:
+    expected = {f"{ground_truth}@{_ref}" for _ref in [ref, sha] if _ref}
+    if signed_claim not in expected:
         raise InvalidPublisherError(
-            f"The claim does not match, expecting {expected!r}, got {signed_claim!r}"
+            f"The claim does not match, expecting one of {sorted(expected)!r}, got "
```

Comment:
Small nitpick, noticed that "the claim" isn't super clear in the error messages we return:

```suggestion
            f"job_workflow_ref does not match, expecting one of {sorted(expected)!r}, got "
```